tools/manifest/vcs.py

```diff
@@ -75,21 +75,24 @@ def __init__(self, root, url_base):
         self.root = root
         self.url_base = url_base
         from gitignore import gitignore
-        self.path_filter = gitignore.PathFilter(self.root)
+        self.path_filter = gitignore.PathFilter(self.root, extras=[".git/"])
 
     def __iter__(self):
-        is_root = True
-        for dir_path, dir_names, filenames in os.walk(self.root):
-            rel_root = os.path.relpath(dir_path, self.root)
-
-            if is_root:
-                dir_names[:] = [item for item in dir_names if item not in
-                                ["tools", "resources", ".git"]]
-                is_root = False
+        paths = list(self.get_paths())
+        for path in paths:
+            if not os.path.isdir(os.path.join(self.root,path)):
+                yield SourceFile(self.root,
+                                 path,
+                                 self.url_base)
 
+    def get_paths(self):
+        path_filter = self.path_filter
+        expanded_path = self.root
+        for dirpath, dirnames, filenames in os.walk(expanded_path):
             for filename in filenames:
-                rel_path = os.path.join(rel_root, filename)
-                if self.path_filter(rel_path):
-                    yield SourceFile(self.root,
-                                     rel_path,
-                                     self.url_base)
+                path = os.path.relpath(os.path.join(dirpath, filename), self.root)
+                if path_filter(path):
+                    yield path
+            dirnames[:] = [item for item in dirnames if
+                           path_filter(os.path.relpath(os.path.join(dirpath, item) + "/",
+                                                       self.root)+"/")]
```

Comment:
The spacing here (and elsewhere) looks inconsistent; there should be spaces around binary operators and after commas.